smartcontract/service/native/ontfs/app_sdk.go

```diff
@@ -0,0 +1,940 @@
+/*
+ * Copyright (C) 2018 The ontology Authors
+ * This file is part of The ontology library.
+ *
+ * The ontology is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ontology is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with The ontology.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ontfs
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/ontio/ontology/common"
+	"github.com/ontio/ontology/common/log"
+	"github.com/ontio/ontology/errors"
+	"github.com/ontio/ontology/smartcontract/service/native"
+	"github.com/ontio/ontology/smartcontract/service/native/utils"
+)
+
+func FsGetNodeInfoList(native *native.NativeService) ([]byte, error) {
+	var nodesInfoList FsNodeInfoList
+
+	source := common.NewZeroCopySource(native.Input)
+	count, err := utils.DecodeVarUint(source)
+	if err != nil {
+		return utils.BYTE_FALSE, errors.NewErr("[APP SDK] FsGetNodeInfoList DecodeVarBytes error!")
+	}
+
+	nodeList := getNodeAddrList(native)
+	if nodeList != nil {
+		r := rand.New(rand.NewSource(time.Now().Unix()))
```

Comment:
nondeterministic execution can easily shutdown the whole network. you can use the current block hash to generate random number.